modin/engines/base/frame/data.py

```diff
@@ -1651,7 +1662,25 @@ def _copartition(self, axis, other, how, sort, force_repartition=False):
         """
         if isinstance(other, type(self)):
             other = [other]
-        if all(o.axes[axis].equals(self.axes[axis]) for o in other):
+
+        is_partitons_aligned = True
+        if (
+            len(self._partitions) != len(other[0]._partitions)
+            and len(self.axes[0]) == len(other[0].axes[0])
+            and axis == 0
+        ):
```

Comment:
`_copartition` function obtains `other` as a list of other frames, so why do we're doing all this only for the first element of a list? Maybe this should be done in a `for` loop which is below?

Reply:
Yes, thanks, this was fixed.